### PR TITLE
Feat: Implement access control table view, search, filtering

### PR DESCRIPTION
**Pull request by AmandaMisjuwar**
## Notion ticket link

[Access control table view](https://www.notion.so/uwblueprintexecs/Access-Management-Table-view-d2fb4cd199a64b929129b79ce72ac00a)
[Access control table search + filtering](https://www.notion.so/uwblueprintexecs/Access-Management-Search-and-Filtering-00671eac6bb248118a7dee283d328616)

## Implementation description
* Get users in API Client
* Display users in a table
* Add search + filter functionality

## Steps to test
1.

## What should reviewers focus on?
* 

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/AccessManagement/index.tsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  const [selectedFilter, setSelectedFilter] = React.useState<Filter>(
+    Filter.ALL,
+  );
+
+  React.useEffect(() => {
+    const getUsers = async () => {
+      const res = await UserAPIClient.getAllUsers();
+      console.log(res);
+      if (res.length !== undefined) setUsers(res);
+      // else setDataError(true);
```
Comment (dhmi19) [inline frontend/src/components/pages/AccessManagement/index.tsx:53]: Is this something we need to be careful about? If not, we can remove the comment I think?

File: frontend/src/components/pages/AccessManagement/index.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    let filteredUsers;
+    if (selectedFilter === Filter.ALL) filteredUsers = users;
+    else if (selectedFilter === Filter.ACTIVE)
+      filteredUsers = users.filter((user) => user.active);
+    else filteredUsers = users.filter((user) => !user.active);
+
+    if (!search) return filteredUsers;
+    return filteredUsers.filter(
+      (user) =>
+        user.email.toLowerCase().includes(search.toLowerCase()) ||
```
Comment (dhmi19) [inline frontend/src/components/pages/AccessManagement/index.tsx:69]: The search bar says to search by name; did we want to search by email ID too? If so, I think we should change the search field to say "Search by name or email"

File: frontend/src/types/AccessManagementTypes.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export interface StatusLabelProps {
```
Comment (dhmi19) [inline frontend/src/types/AccessManagementTypes.ts:1]: Should we rename this to UserStatusLabelProps?

**Review by dhmi19 — COMMENTED** [review]
Looks good to me overall. Just left a couple of comments with some questions, but should be quick fixes or specific nit picks which we don't have to change

File: frontend/src/components/pages/AccessManagement/index.tsx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    ACTIVE = "Active",
+    INACTIVE = "Inactive",
+  }
+
+  enum UserRoles {
+    ADMIN = "FON Admin",
+    CAMP_COORDINATOR = "Camp Coordinator",
+  }
+
+  const filterOptions = [Filter.ALL, Filter.ACTIVE, Filter.INACTIVE];
```
[the remark this answers is not in the document]
Comment (AmandaMisjuwar) [reply frontend/src/components/pages/AccessManagement/index.tsx:39]: I feel like writing it out is more readable/quick to understand and I don't think there will be other filter options being added but I can change if rly needed!

File: frontend/src/components/common/NavBar.tsx [hunk 5]
```diff
@@ -47,7 +47,7 @@ const NavBar = (): JSX.Element => {
         history.push(Routes.GLOBAL_FORMS_PAGE);
         break;
       case 2:
-        history.push(Routes.ACCESS_CONTROL_PAGE);
+        history.push(Routes.ACCESS_MANAGEMENT_PAGE);
```
[the remark this answers is not in the document]
Comment (AmandaMisjuwar) [reply frontend/src/components/common/NavBar.tsx]: Changed it back to Access Control

**Review by annechung — APPROVED** [review]

**Pull request merged**